File: SDKs/googleapis.com/dataflow/v1b3/python/src/sdk/models/shared/runtimemetadata.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class RuntimeMetadata:
    r"""RuntimeMetadata
    RuntimeMetadata describing a runtime environment.
    """
    
    parameters: Optional[List[ParameterMetadata]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('parameters') }})
    sdk_info: Optional[SDKInfo] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sdkInfo') }})
    

File: SDKs/googleapis.com/dataflow/v1b3/typescript/src/sdk/models/shared/runtimemetadata.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ParameterMetadata } from "./parametermetadata";
import { SDKInfo } from "./sdkinfo";



// RuntimeMetadata
/** 
 * RuntimeMetadata describing a runtime environment.
**/
export class RuntimeMetadata extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=parameters", elemType: ParameterMetadata })
  parameters?: ParameterMetadata[];

  @SpeakeasyMetadata({ data: "json, name=sdkInfo" })
  sdkInfo?: SDKInfo;
}
